### ISSUE TITLE
Starting issue with electron

### ISSUE THREAD

**Issue by IonutBurlacu** [pokusew/nfc-pcsc#48]
Hello. I'm developing a React+Electron app and I'm also using this library to read some bands with a nfc reader. The problem I have is that on some computers the app runs perfectly, but on the others, the app starts as a background process and it stays there with no error.
Any idea what can cause this to happen?
![46721354-76bde400-cc7b-11e8-9e98-01bd1a694fe1](https://user-images.githubusercontent.com/6929354/46904884-2b623a80-cef4-11e8-959e-449c492c486a.png)

**Comment by danielhanzs** [comment]
Attach a card-reader to the computer then start the program, it issome logic problem that if no card reader found it freezes the program withput any error response.

**Comment by IonutBurlacu** [comment]
@danielhanzs I tried what you said. I installed the needed usb driver and attached the usb reader and now the app works. The strange thing is that it works now even without the nfc reader connected. Like it installed something when it connected and now it doesn't need it anymore. Any idea how to solve this?

**Comment by IonutBurlacu** [comment]
I have another person which tried only installing the nfc usb driver, but no success. So it must be related to the usb reader itself.

**Comment by danielhanzs** [comment]
If you connect the device and start the application and desconnect the devie without restarting the application it will start searching for the readers instead of stopping the application which is good.

I will try tomorrow the https://www.npmjs.com/package/usb-detection repo

var usbDetect = require('usb-detection');
usbDetect.startMonitoring();
 
usbDetect.on('add', function(device) {
    console.log(device);
});

On add event I will check if the device, vendorId or productId or deviceName or manufacturer is the one I'm searching for (needed only for ACR122 USB reader) and then I will call the new NFC() class.

I'm at home will test it tomorrow at workplace if physical reader needed to be connected or the installed driver will solve the problem.

For a solid solution I will stick with usb detect.

**Comment by IonutBurlacu** [comment]
Ok. Thank you very much!

**Comment by TeodorDre** [comment]
> If you connect the device and start the application and desconnect the devie without restarting the application it will start searching for the readers instead of stopping the application which is good.
> 
> I will try tomorrow the https://www.npmjs.com/package/usb-detection repo
> 
> var usbDetect = require('usb-detection');
> usbDetect.startMonitoring();
> 
> usbDetect.on('add', function(device) {
> console.log(device);
> });
> 
> On add event I will check if the device, vendorId or productId or deviceName or manufacturer is the one I'm searching for (needed only for ACR122 USB reader) and then I will call the new NFC() class.
> 
> I'm at home will test it tomorrow at workplace if physical reader needed to be connected or the installed driver will solve the problem.
> 
> For a solid solution I will stick with usb detect.
> 
> PS.: For usb-detection you need full vs2017 community the windows-build-tools not sufficient enough to build it.

Hello! it is actually now, does it was succeed?  on 50% of windows computer, i have a problem with it

**Comment by CoWayger** [comment]
I don't think this is the "right" solution, there are many card readers on market and manually listing them in some enum is not very maintainable approach. Is there some flag for "not waiting for card reader connection"?

**Comment by liamcmitchell** [comment]
I can reproduce this by disabling the SCardSrv service...

<img width="517" height="62" alt="Image" src="https://github.com/user-attachments/assets/5e878217-dcb9-4e6d-8147-6a1f83dd7857" />

...running my electron app and then clicking "No" in the permission prompt:

<img width="450" height="370" alt="Image" src="https://github.com/user-attachments/assets/dc5106cb-03ef-4f98-8246-35cfa09569ae" />

The program prints `Shell Execute failed with 1223` to the console and then freezes.

I think it's this block of code and the freeze is being stuck in the loop on line 72:

https://github.com/pokusew/node-pcsclite/blob/8100373ef05278b274250695577584fb161dec9b/src/pcsclite.cpp#L49-L78

Workaround is to check service is running before init:
```js
import { NFC }  from "nfc-pcsc";
import { promisify } from 'node:util';
import child_process from 'node:child_process';
const execAsync = promisify(child_process.exec);

export async function startNfc() {
  if (process.platform === "win32") {
    // Checking if Smart Card service (SCardSvr) is running
    const { stdout: serviceStatus } = await execAsync("sc query SCardSvr");
    if (!/RUNNING/.test(serviceStatus)) {
      // Checking if Smart Card service (SCardSvr) is configured to auto-start
      const { stdout: serviceStart } = await execAsync(
        `$service = Get-ItemProperty -Path HKLM:\\System\\CurrentControlSet\\Services\\SCardSvr; $service.Start`,
        {
          shell: "powershell.exe",
        },
      );
      if (serviceStart.trim() !== "2") {
        // Smart Card service (SCardSvr) is not running and not configured to auto-start, requesting administrator privileges to configure auto-start
        await execAsync(
          `Start-Process -Verb RunAs -WindowStyle Hidden -PassThru -Wait -FilePath sc.exe -ArgumentList "config SCardSvr start=auto"`,
          {
            shell: "powershell.exe",
          },
        );
      }
    }
  }

  const nfc = new NFC();
  // ...
}
```

**Comment by robincsamuel** [comment]
I'm having the same issue. The app becomes unresponsive on Windows when calling `new NFC()`. It works fine if I comment it out. 

@pokusew , it would be great if you could share your thoughts.

**Comment by robincsamuel** [comment]
I’m experiencing the same issue on Windows machines. The Electron app becomes unresponsive when `new NFC()` is called. To avoid this, I’m currently checking the `SCardSvr` status. If the status is stopped, I don’t initialise `NFC`.

@pokusew is there a better way to handle this?

**Comment by vmangelovv** [comment]
Any progress on this ? @pokusew
